### PR TITLE
Avoid redundant authz indices contains check for wildcard expansion

### PR DESCRIPTION
**Pull request by albertzaharovits**
The resolved indices list contains name elements that either:
* are concrete names that been explicitly mentioned in the original request, or
* are authorized indices that are referred to by a date math expression from the request, or
* are authorized indices that are covered by a wildcard from the request

If the request option `ignoredUnavailable` is set to `true` the resolved indices list overall
must silently ignore the names that are not in the authorized set.
Consequently, only the names in the first category mentioned above must be verified;
the check for the other names is redundant.

Related https://github.com/elastic/elasticsearch/pull/76481

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstractionResolver.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public IndexAbstractionResolver(IndexNameExpressionResolver indexNameExpressionR
             } else if (dateMathName.equals(indexAbstraction)) {
```
Comment (albertzaharovits) [inline server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstractionResolver.java:103]: Note that the current behavior in Security is different the "core" behavior.
In core, excluding missing index names (ie "-nonexistent") is an error, but not in this case.
Most likely, the fix for it is to adopt the same behavior from the date math expression evaluation from a couple of lines above.

But, I'll stay away from it for now.
The robust, non-patchy fix, is to inject into the index name resolver from core the authorized indices view.

The proposed fix in this PR maintains the current (buggy) behavior, but avoids redundant `availableIndexAbstractions.contains` checks, hopefully having a small positive performance impact.

**Comment by albertzaharovits** [comment]
@elasticmachine test this

**Comment by albertzaharovits** [comment]
@elasticmachine test this please

**Review by ywangd — APPROVED** [review]
LGTM

Thanks Albert!

**Review by tvernum — APPROVED** [review]
LGTM from a security point of view, but I think we need someone from Core/Infra (or Data Mgt) to review the change to `IndexAbstractionResolver`

It worries me that we're making this sort of change to a class in core, but didn't need to change any tests  - that is, there are no existing tests for how that class should deal with missing concrete indices.

**Comment by albertzaharovits** [comment]
> It worries me that we're making this sort of change to a class in core, but didn't need to change any tests - that is, there are no existing tests for how that class should deal with missing concrete indices.

This method is ultimately only used in the `indices:admin/resolve/index` action, which ignores unknown indices anyway: https://github.com/elastic/elasticsearch/blob/9c5f148624b1e1a6a4c627fe06c320eebf9c6f69/server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java#L537 
I had to triple check.

**Review by jbaiera — APPROVED** [review]
LGTM

**Pull request merged**